app/components/join-osf-banner/component.ts

```diff
@@ -0,0 +1,26 @@
+import { action, computed } from '@ember-decorators/object';
+import { service } from '@ember-decorators/service';
+import Component from '@ember/component';
+import { SessionService } from 'ember-simple-auth/addon/services/session';
+import Analytics from '../../services/analytics';
+
+export default class JoinOsfBanner extends Component.extend() {
+    @service analytics: Analytics;
+    @service session: SessionService;
+
+    storage = window.localStorage;
```

Comment:
May want to consider https://github.com/funkensturm/ember-local-storage, which provides a nice Embery abstraction.

Reply:
I thought about it, but in the spirit of not adding unnecessary dependencies, I though I'd skip it. We are literally only checking to see if a key exists and setting it if it isn't, and (aside from ember-simple-auth), this is the only use of local storage on the OSF in 5 years.